bluemira/equilibria/analysis.py

```diff
@@ -0,0 +1,1341 @@
+        lgth, ang = get_legs_length_and_angle(eq, legs, pfb)
+        if legs_to_plot in DivLegsToPlot.PAIR:
+            location = "lower" if legs_to_plot is DivLegsToPlot.LW else "upper"
+            if f"{location}_inner" not in lgth:
+                raise BluemiraError(
+                    f"One of your chosen equilibria does not have {location}_inner legs."
+                )
+            if f"{location}_outer" not in lgth:
+                raise BluemiraError(
+                    f"One of your chosen equilibria does not have {location}_outer legs."
+                )
+            lgth = {k: lgth[k] for k in (f"{location}_inner", f"{location}_outer")}
+            ang = {k: ang[k] for k in (f"{location}_inner", f"{location}_outer")}
+        else:
+            # sort alphabetically
+            lgth = {k: lgth[k] for k in sorted(lgth)}
+            ang = {k: ang[k] for k in sorted(ang)}
+        lengths.append(lgth)
+        angles.append(ang)
+    return lengths, angles
+
+
+def get_target_flux(eq, target, target_coords, n_layers, vertical=False):  # noqa: FBT002
+    """
+    Get a selection flux surfaces that cross the divertor target.
+
+    eq:
+        chosen Equilibrium
+    target:
+        Location of divertor target, lower_outer, lower_inner, upper_outer or upper_inner
+    target_coords:
+        Corrinates of divertor target
+    n_layers:
+        Number of flux surfaces to extract for each leg
+    vertical:
+        Set to True if using a vertical target.
+        This boolean determines whether to get flux surfaces across
+        the target x-range (horizontal) or z-range (vertical).
+
+    Raises
+    ------
+    BluemiraError
+        If the target is set to vertical or horizontal incorectly.
+
+    Returns
+    -------
+    fs_list:
+        List of flux surface coordinates.
+
+    """
+    if eq._o_points is None:
+        _, _ = eq.get_OX_points()
+
+    if not vertical:
+        x = np.min(target_coords.x)
+        z = target_coords.z[np.argmin(target_coords.x)]
+        target_size = np.abs(np.max(target_coords.x) - np.min(target_coords.x))
+        if target_size == 0:
+            raise BluemiraError(
+                "No x-range found for target coords,"
+                " perhaps you are using a vertical target (set vertical=True)."
+            )
+        target_offsets = np.linspace(0, target_size, n_layers)[1:]
+        dx = x + target_offsets
+        dz = np.ones(n_layers) * z
+    else:
+        x = target_coords.z[np.argmin(target_coords.z)]
+        z = np.min(target_coords.z)
+        target_size = np.abs(np.max(target_coords.z) - np.min(target_coords.z))
+        if target_size == 0:
+            raise BluemiraError(
+                "No z-range found for target coords,"
+                " perhaps you are using a horizontal target (set vertical=False)."
+            )
+        target_offsets = np.linspace(0, target_size, n_layers)[1:]
+        dx = np.ones(n_layers) * x
```

Comment:
There is a dz above too (I may have got the arguments the wrong way around)
```suggestion
        dx = np.full(n_layers, x)
```